numpy/testing/tests/test_utils.py

```diff
@@ -1114,16 +1114,30 @@ def test_simple(self):
                           lambda: assert_string_equal("foo", "hello"))
 
 
-def assert_warn_len_equal(mod, n_in_context, py3_n_in_context=None):
+def assert_warn_len_equal(mod, n_in_context, py3_n_in_context=None, other=False):
     mod_warns = mod.__warningregistry__
     # Python 3.4 appears to clear any pre-existing warnings of the same type,
     # when raising warnings inside a catch_warnings block. So, there is a
     # warning generated by the tests within the context manager, but no
     # previous warnings.
     if 'version' in mod_warns:
+        # Python 3.5+ adds a 'version' entry to the registry
         if py3_n_in_context is None:
             py3_n_in_context = n_in_context
-        assert_equal(len(mod_warns) - 1, py3_n_in_context)
+        if sys.version_info[:2] < (3, 7):
+            # Do not count the 'version' entry
+            assert_equal(len(mod_warns) - 1, py3_n_in_context)
```

Comment:
Can you add `n_warns = len(mod_warns) - 1`, and then use that below?

Reply:
I rewrote the whole bit to be simpler. Added two keywords `py34` and `py37` to account for the version dependent changes. Python 3.7 does not make entries for ignored warnings, so I suspect we will no longer need `suppress_warnings` when we drop support for earlier Python versions.